### PR TITLE
[Organizations] Obfuscate organization routes containing usernames

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/NuGet/NuGetGallery/issues/5676

I introduced a couple routes containing usernames (mostly for handling membership/transformation requests). This work obfuscates the username in these routes and also tidies up the routing in general.

### CODE REVIEW

**Comment by scottbommarito** [comment]
@joelverhagen https://github.com/NuGet/NuGetGallery/issues/5678

File: src/NuGetGallery/App_Start/Routes.cs [hunk 1]
```diff
@@ -309,32 +336,96 @@ public static void RegisterUIRoutes(RouteCollection routes)
...
+                new RouteExtensions.ObfuscatedMetadata(1, Obfuscator.DefaultTelemetryUserName));
+
+            routes.MapRoute( 
+                RouteName.OrganizationMemberCancelAjax,
+                "organization/{accountName}/members/cancel",
+                new { controller = "Organizations", action = RouteName.OrganizationMemberCancelAjax },
+                new[]
+                {
+                    new RouteExtensions.ObfuscatedMetadata(1, Obfuscator.DefaultTelemetryUserName),
+                    new RouteExtensions.ObfuscatedMetadata(4, Obfuscator.DefaultTelemetryUserName)
```
Comment (chenriksson) [inline src/NuGetGallery/App_Start/Routes.cs]: Do you need to obfuscate index 4?

File: tests/NuGetGallery.Facts/Extensions/RouteExtensionsFacts.cs [hunk 2]
```diff
@@ -22,16 +22,16 @@ public void MapRouteAddObfuscation()
 
             // Act + Assert
             Assert.True(RouteExtensions.ObfuscatedRouteMap.ContainsKey(_routeUrl));
-            Assert.Equal(_segment, RouteExtensions.ObfuscatedRouteMap[_routeUrl].ObfuscatedSegment);
-            Assert.Equal(_obfuscatedValue, RouteExtensions.ObfuscatedRouteMap[_routeUrl].ObfuscateValue);
+            Assert.Equal(_segment, RouteExtensions.ObfuscatedRouteMap[_routeUrl][0].ObfuscatedSegment);
+            Assert.Equal(_obfuscatedValue, RouteExtensions.ObfuscatedRouteMap[_routeUrl][0].ObfuscateValue);
```
Comment (chenriksson) [inline tests/NuGetGallery.Facts/Extensions/RouteExtensionsFacts.cs]: No other test updates? I thought we had tests for the routes being obfuscated...

File: src/NuGetGallery/App_Start/Routes.cs (hunk 1 above)
Comment (scottbommarito) [reply src/NuGetGallery/App_Start/Routes.cs]: woops, good catch

File: tests/NuGetGallery.Facts/Extensions/RouteExtensionsFacts.cs (hunk 2 above)
Comment (scottbommarito) [reply tests/NuGetGallery.Facts/Extensions/RouteExtensionsFacts.cs]: The test for the routes being obfuscated was just checking https://github.com/NuGet/NuGetGallery/pull/5677/files#diff-33320da487df9e20194b38d94d65876aR17

**Pull request merged**